doc/yaml_configuration.md

```diff
@@ -427,6 +427,57 @@ compiler: ghcjs-0.1.0.20150924_ghc-7.10.2
 compiler-check: match-exact
 ```
 
+### compiler-build (experimental)
+
+(Since 1.10.0)
+
+Build a GHC compiler from source during the "setup" phase. The version to build
+is defined by a Git commit id and an Hadrian flavour (Hadrian is the build
+system of GHC). For instance:
+
+```yaml
+compiler-build:
+  git: https://gitlab.haskell.org/ghc/ghc.git
+  commit: 5be7ad7861c8d39f60b7101fd8d8e816ff50353a
+  flavour: quick
+```
+
+To use a built compiler you have to select it by its name which has the the
+following form: "ghc-git-COMMIT-FLAVOUR". For example:
+
+```yaml
+compiler: ghc-git-5be7ad7861c8d39f60b7101fd8d8e816ff50353a-quick
+```
+
+#### Global packages
+
+The GHC compiler you build from sources may depend on unreleased versions of
+some global packages (e.g. Cabal). It may be an issue if a package you try to
+build with this compiler depends on such global packages because Stack may not
+be able to find versions of those packages (on Hackage, etc.) that are
+compatible with the compiler.
+
+The easiest way to deal with this issue is to add the relevant packages as
```

Comment:
Done. Not tested yet though.

I was thinking, maybe it would be clearer to have something like this to mean the same thing:

```yaml
extra-deps:
- global: Cabal
```

Because we are not really dropping a package in this case: we drop it from the resolver to have access to the "global" one (i don't like "global" either; maybe "bundled" or something else could be better).

Reply:
I don't want to add additional syntax for this case.